files/class-vip-filesystem-api-cache.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Automattic\VIP\Files;
+
+use WP_Error;
+
+
+class API_Cache {
+
+	/**
+	 * @var API_Client  Holds the class instance
+	 */
+	private static $instance = null;
+
+	/**
+	 * @var array   Array of created local cache files
+	 */
+	private $files = [];
+
+	/**
+	 * @var string  Temp directory to cache file in
+	 */
+	private $tmp_dir = '/tmp';
+
+	/**
+	 * API_Cache constructor.
+	 */
+	protected function __construct() {
+		// sys_get_temp_dir is pretty inconsistent regarding trailing slashes
+		$this->tmp_dir = rtrim( sys_get_temp_dir(), '/' );
```

Comment:
Maybe better to use the WordPress `get_temp_dir` function? It also returns a consistent slash so we may want to update other uses to take that into account.

```suggestion
		$this->tmp_dir = get_temp_dir();
```

https://developer.wordpress.org/reference/functions/get_temp_dir/

Reply:
Ah I did not know about that function. It's probably a better idea to use it then.